### ISSUE TITLE
Road to basic Linux compatibility

### ISSUE THREAD

**Issue by felix91gr** [krzysztofzablocki/Sourcery#630]
I want to try and make a PR that lets you compile Sourcery and get basic, on-demand, templating working. 

This idea has two goals:
a) Make Sourcery-based projects available for Linux
b) Make generation of the `LinuxMain.swift` file possible on Linux. 

How should I start? @mxcl gave some hints at #306 but I don't really know how to make that more concrete.

**Comment by mxcl** [comment]
I just pushed what work I did to my fork.

**Comment by felix91gr** [comment]
Thank you!

What's left to do in order to get a working "baseline-Sourcery"?

I got this output:

<details>
<summary>Output of <code>swift build</code></summary>

```console
Fetching https://github.com/kylef/Commander.git
Fetching https://github.com/kylef/PathKit.git
Fetching https://github.com/jpsim/SourceKitten.git
Fetching https://github.com/IBM-Swift/CommonCrypto.git
Fetching https://github.com/SwiftGen/StencilSwiftKit.git
Fetching https://github.com/xcodeswift/xcproj.git
Fetching https://github.com/drmohundro/SWXMLHash.git
Fetching https://github.com/kylef/Spectre.git
Fetching https://github.com/kylef/Stencil.git
Fetching https://github.com/tadija/AEXML.git
Fetching https://github.com/JohnSundell/ShellOut.git
Fetching https://github.com/antitypical/Result.git
Fetching https://github.com/norio-nomura/Clang_C.git
Fetching https://github.com/norio-nomura/SourceKit.git
Fetching https://github.com/jpsim/Yams.git
Fetching https://github.com/tomlokhorst/XcodeEdit.git
Fetching https://github.com/Carthage/Commandant.git
Fetching https://github.com/seanparsons/SwiftTryCatch.git
Cloning https://github.com/seanparsons/SwiftTryCatch.git
Resolving https://github.com/seanparsons/SwiftTryCatch.git at e7074a72e4d4dc516f391bc4d4afd8ca6a845b4b
Cloning https://github.com/Carthage/Commandant.git
Resolving https://github.com/Carthage/Commandant.git at 0.12.0
Cloning https://github.com/drmohundro/SWXMLHash.git
Resolving https://github.com/drmohundro/SWXMLHash.git at 4.2.5
Cloning https://github.com/kylef/Spectre.git
Resolving https://github.com/kylef/Spectre.git at 0.7.2
Cloning https://github.com/IBM-Swift/CommonCrypto.git
Resolving https://github.com/IBM-Swift/CommonCrypto.git at 0.1.5
Cloning https://github.com/JohnSundell/ShellOut.git
Resolving https://github.com/JohnSundell/ShellOut.git at 2.1.0
Cloning https://github.com/kylef/PathKit.git
Resolving https://github.com/kylef/PathKit.git at 0.8.0
Cloning https://github.com/norio-nomura/Clang_C.git
Resolving https://github.com/norio-nomura/Clang_C.git at 1.0.2
Cloning https://github.com/kylef/Stencil.git
Resolving https://github.com/kylef/Stencil.git at 0.9.0
Cloning https://github.com/jpsim/SourceKitten.git
Resolving https://github.com/jpsim/SourceKitten.git at 0.18.4
Cloning https://github.com/jpsim/Yams.git
Resolving https://github.com/jpsim/Yams.git at 0.4.1
Cloning https://github.com/antitypical/Result.git
Resolving https://github.com/antitypical/Result.git at 3.2.4
Cloning https://github.com/tadija/AEXML.git
Resolving https://github.com/tadija/AEXML.git at 4.2.2
Cloning https://github.com/SwiftGen/StencilSwiftKit.git
Resolving https://github.com/SwiftGen/StencilSwiftKit.git at 2.3.0
Cloning https://github.com/xcodeswift/xcproj.git
Resolving https://github.com/xcodeswift/xcproj.git at 4.0.0
Cloning https://github.com/kylef/Commander.git
Resolving https://github.com/kylef/Commander.git at 0.6.1
Cloning https://github.com/norio-nomura/SourceKit.git
Resolving https://github.com/norio-nomura/SourceKit.git at 1.0.1
Cloning https://github.com/tomlokhorst/XcodeEdit.git
Resolving https://github.com/tomlokhorst/XcodeEdit.git at 1.1.0
error: could not find target(s): Sourcery; use the 'path' property in the Swift 4 manifest to set a custom target path
```
</details>

**Comment by felix91gr** [comment]
> it looks like you just need to change base to NSObject

I understood this on principle, but I looked up those classes and all of them directly or transitively inherit from NSObject. I must be missing something :thinking: 

>  the error messages here are looking normal, those changes are just required with newer toolchain

Which toolchain should I use? I was using Swift 4.0 to compile these.

**Comment by art-divin** [comment]
This is implemented with #1188
